### PR TITLE
Regression test `println!()` panic message on `ErrorKind::BrokenPipe`

### PR DESCRIPTION
**Pull request by Enselic**
No existing test (that I could find) failed if the `panic!()` of the `println!()` family of functions was removed, or if its message was changed:

https://github.com/rust-lang/rust/blob/104f4300cfddbd956e32820ef202a732f06ec848/library/std/src/io/stdio.rs#L1007-L1009

So add such a test.

This is in preparation of adding a hint about the existence of [`unix_sigpipe`](https://github.com/rust-lang/rust/issues/97889) if that is the reason for the panic.

Even if we don't end up adding a hint, this is still a sensible test to have, I think.

@rustbot label +A-testsuite +A-io +T-libs +O-unix

### CODE REVIEW

**Comment by Enselic** [comment]
I'm pretty sure T-libs is much for appropriate for this PR than T-compiler.

@rustbot label -T-compiler

r? T-libs

File: tests/ui/process/println-and-broken-pipe.run.stderr [hunk 1]
```diff
@@ -0,0 +1,2 @@
+thread 'main' panicked at 'failed printing to stdout: Broken pipe (os error 32)', library/std/src/io/stdio.rs:1008:9
```
Comment (Enselic) [inline tests/ui/process/println-and-broken-pipe.run.stderr:1]: We'll most likely want to filter out or wildcard `library/std/src/io/stdio.rs:1008:9` somehow. Before I figure out how to do that I wanted to check what CI thinks of my PR, and if there are any early code review comments.

Comment (Enselic) [reply tests/ui/process/println-and-broken-pipe.run.stderr:1]: Actually, CR can wait until I have figured out what to do here.

@rustbot author

Comment (Nilstrieb) [reply tests/ui/process/println-and-broken-pipe.run.stderr:1]: you want to look into the `// normalize-stderr` directive (by looking at other tests that use it, i dont think there are any docs for compiletest lol)

**Comment by Nilstrieb** [comment]
r? libs

**Comment by Enselic** [comment]
CI passed and the code and its output looks good to me. This is ready for review.

@rustbot ready

@rustbot label -A-testsuite

**Comment by Enselic** [comment]
@joshtriplett Friendly 7-week review ping. There is no urgency to review this; I just want to double-check that this PR has not gone under your radar :)

Elevator pitch for this PR: I want to make the `println!()` panic message on `BrokenPipe` suggest the use of [`unix_sigpipe`](https://github.com/rust-lang/rust/issues/97889). But before I make any changes I want there to be a regression test in place for the current behavior. This PR is that regression test.

**Review by tgross35 — APPROVED** [review]
LGTM, thorough test

**Comment by workingjubilee** [comment]
or: looks good to me.
r? @workingjubilee

**Comment by Enselic** [comment]
@workingjubilee I pushed a fix for the [CI failure](https://github.com/rust-lang/rust/pull/114085#issuecomment-1651604415). I had to add `.env("RUST_BACKTRACE", "0")` to the test child process. I could reproduce the CI failure locally with `RUST_BACKTRACE=1 ./x test --force-rerun tests/ui/process/println-with-broken-pipe.rs` and this fix worked locally, so will probably also work in CI.

**Comment by workingjubilee** [comment]
Thank you for being so quick to look into that! I am going to give this a slightly closer look to verify before r+ing again.

File: tests/ui/process/println-with-broken-pipe.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+#![feature(unix_sigpipe)]
+
+use std::env;
+use std::process::{Command, Stdio};
+
+#[unix_sigpipe = "sig_ign"]
+fn main() {
+    let mut args = env::args();
+    let me = args.next().unwrap();
```
Comment (workingjubilee) [inline tests/ui/process/println-with-broken-pipe.rs:21]: ...cute.

File: tests/ui/process/println-with-broken-pipe.rs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        // loop will end with a panic more or less immediately.
+        for _ in 0..65536 * 64 {
+            println!("{arg}");
+        }
+        unreachable!("should have panicked because of BrokenPipe");
+    }
+
+    // Set up a pipeline with a short-lived consumer and wait for it to finish.
+    // This will produce the `println!()` panic message on stderr.
+    let mut producer = Command::new(&me)
```
Comment (workingjubilee) [inline tests/ui/process/println-with-broken-pipe.rs:34]: Extremely cute.

**Review by workingjubilee — APPROVED** [review]

**Comment by workingjubilee** [comment]
I feel confident this will pass CI this time, however,
@bors rollup=iffy

**Pull request merged**